tests/mock_crust/utils.rs

```diff
@@ -621,22 +608,134 @@ pub fn add_connected_nodes_until_split(
+    prefixes_to_add_to_split
+}
+
+// This adds new nodes until we reach the requested size for each prefix. No split expected.
+fn add_connected_nodes_until_sized(
+    network: &Network<PublicId>,
+    nodes: &mut Vec<TestNode>,
+    prefixes_new_count: &[PrefixAndSize],
+    use_cache: bool,
+) {
+    clear_all_event_queues(nodes, |_| {});
+
+    add_nodes_to_prefixes(network, nodes, prefixes_new_count, use_cache);
+
+    clear_all_event_queues(nodes, |_| {});
+    clear_relocation_overrides(nodes);
+
+    trace!(
+        "Filled prefixes until ready to split {:?}",
+        prefixes_new_count
+    );
+}
+
+// Start the target number of new nodes under each target prefix.
+fn add_nodes_to_prefixes(
+    network: &Network<PublicId>,
+    nodes: &mut Vec<TestNode>,
+    prefixes_new_count: &[PrefixAndSize],
+    use_cache: bool,
+) {
+    let mut rng = network.new_rng();
+
+    for (prefix, target_count) in prefixes_new_count {
+        let num_in_section = nodes
+            .iter()
+            .filter(|node| prefix.matches(&node.name()))
+            .count();
+        // To ensure you don't hit this assert, don't have more than `min_split_size()` entries in
+        // `nodes` when calling this function.
+        assert!(
+            num_in_section <= *target_count,
+            "The existing nodes' names disallow creation of the requested prefixes. There \
+             are {} nodes which all belong in {:?} which exceeds the limit here of {}.",
+            num_in_section,
+            prefix,
+            target_count
+        );
+        let to_add_count = target_count - num_in_section;
+        for _ in 0..to_add_count {
+            add_node_to_section(network, nodes, prefix, &mut rng, use_cache);
+        }
+    }
+}
+
+// Clear all event queues applying check_event to them.
+fn clear_all_event_queues(nodes: &mut Vec<TestNode>, check_event: impl Fn(Event)) {
     for node in nodes.iter_mut() {
         while let Ok(event) = node.try_next_ev() {
-            match event {
-                Event::NodeAdded(..)
-                | Event::NodeLost(..)
-                | Event::TimerTicked
-                | Event::SectionSplit(..) => (),
-                event => panic!("Got unexpected event: {:?}", event),
-            }
+            check_event(event)
         }
+    }
+}
+
+// Clear all `next_relocation_dst` / `next_relocation_interval` values.
+pub fn clear_relocation_overrides(nodes: &mut Vec<TestNode>) {
+    for node in nodes.iter_mut() {
         node.inner.set_next_relocation_dst(None);
         node.inner.set_next_relocation_interval(None);
     }
+}
 
-    trace!("Created testnet comprising {:?}", prefixes);
+// Returns how many nodes to add to each sub-prefixes to reach split with one extra node.
+// The second field of the return value contain the sub-prefixes to add the final node to split.
+fn prefixes_and_count_to_split_with_only_one_extra_node(
```

Comment:
I think the returned value in `Vec<PrefixAndSize>` is the `expected target section size`, not `how many nodes to add` ?

Reply:
Correct, will fix.